### ISSUE TITLE
Sliderb.png skin element won't show up

### ISSUE THREAD

**Issue by Exi32** [ppy/osu#16007]
With one specific skin the sliderb.png element won't show up with some sliders, that always happen to be the same ones. I attached the map from the video as an example, it does this on a friend's client too.

Video: https://drive.google.com/file/d/1NiAr6zfoDebq7Enj7dUKKjx0CQCVbSXG/view
Skin: https://www.mediafire.com/file/tjub797ui36nrrt/operation+villa+16_9+-+yugen+cursor.osk/file
Map: https://osu.ppy.sh/beatmapsets/857624#osu/1791905

**Comment by bdach** [comment]
the skin linked has a broken combo colour with zero alpha:

```
Combo1: 0,120,255
Combo2: 60,255,60
Combo3: 120,0,255,0 // <- this one
```

this is guarded against in several places by using [a special legacy compatibility function](https://github.com/ppy/osu/blob/b1d78c4284f05a3181832138e9a4e4109fed4a6e/osu.Game/Skinning/LegacyColourCompatibility.cs#L22), but unfortunately with the way the combo colour works (via `IHasAccentColour`) that really can't be applied directly (non-legacy paths shouldn't call legacy functions).

maybe it is feasible to force zero alpha here, possibly?

https://github.com/ppy/osu/blob/b1d78c4284f05a3181832138e9a4e4109fed4a6e/osu.Game/Skinning/LegacySkin.cs#L133-L134

worried that will break something somewhere, though. probably the safest thing to do is to just tell users to fix their skins?

**Comment by smoogipoo** [comment]
I believe it will break stuff, though I'm not 100% sure. It's why I've chosen to fix it locally in cases like this. Is there a reason it can't be done here?

**Comment by bdach** [comment]
The ball receives its colour in the following manner:

https://github.com/ppy/osu/blob/69fc7782ecd7a4279feca2b5f34b39f4163f7d0f/osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSlider.cs#L267

So fixing this locally would require the DHO `AccentColour` application paths to call `DisallowZeroAlpha`. I'm not sure that is fine. I recall the rule was that non-legacy DHOs shouldn't call legacy hack functions like that one.

**Comment by smoogipoo** [comment]
Yeah, or implement `IHasAccentColour` on both legacy/default components...
